modules/consensus/synchronize.go

```diff
@@ -11,47 +11,35 @@ import (
 )
 
 const (
-	MaxCatchUpBlocks = 50
+	MaxCatchUpBlocks       = 50
+	MaxSynchronizeAttempts = 8
+	ResynchronizeTimeout   = time.Second * 20
 )
 
-// threadedResynchronize continuously calls Synchronize on a random peer every
-// few minutes. This helps prevent unintentional desychronization in the event
-// that broadcasts start failing.
+// threadedResynchronize will call synchronize on up to 8 random peers.
 func (s *State) threadedResynchronize() {
 	for {
-		go func() {
-			// The set of connected peers is small and randomly ordered, so
-			// just naively iterate through them until a Synchronize succeeds.
-			for _, peer := range s.gateway.Peers() {
-				if s.Synchronize(peer) == nil {
-					break
-				}
+		peers := s.gateway.Peers()
+		for _, peer := range peers {
+			// NOTE: error is not checked, because nothing happens whether
+			// there is an error or not, the node continuously synchronizes
+			// to all peers.
+			err := s.Synchronize(peer)
+			if err != nil {
+				continue
 			}
-		}()
-		time.Sleep(time.Minute * 2)
+			time.Sleep(ResynchronizeTimeout)
+		}
+		time.Sleep(ResynchronizeTimeout)
```

Comment:
my understanding of the loop is that it will synchronize to up to 8 peers, sleeping for 20s in between each synchronize, and then sleep for an additional 20s. I'm suggesting that the latter sleep be a few minutes instead of 20s.